votepy/chamberlin-courant.py

```diff
@@ -67,6 +68,100 @@ def scoring_function(committee: Iterable[int], voting: OrdinalElection, candidat
                   lambda committee, voting, candidate: scoring_function(committee, voting, candidate,
                                                                         number_of_scored_candidates))
 
+def chamberlin_courant_ilp(voting: Union[OrdinalElection, list[int]], size_of_committee: int,
+                              number_of_scored_candidates: int, solver: Union[Type[Gurobi], Type[CPLEX]] = Gurobi) -> list[int]:
+    """Implementation of the chamberlin-courant rule, using ILP formulation by:
+    Peters, Dominik & Lackner, Martin. (2020).
+    Preferences Single-Peaked on a Circle.
+    Journal of Artificial Intelligence Research. 68. 463-502. 10.1613/jair.1.11732.
+
+    Args:
+        voting (Union[OrdinalElection, list[int]]): Voting for which the function calculates the committee
+        size_of_committee (int): Size of the committee
+        number_of_scored_candidates (int): Number of scored candidartes using k-borda rule
+    Returns:
+        list[int]: List of chosen candidates
+    """
+    if not isinstance(voting, OrdinalElection):
+        voting = OrdinalElection(voting)
+
+    model = solver()
+    x = []
+    for i in range(voting.number_of_voters):
+        x.append([])
+        for r in range(voting.ballot_size):
+            x[-1].append(model.addVariable(f"x_{i},{r}", 'B', 1))
+
+    y = []
+    for c in range(voting.ballot_size):
+        y.append(model.addVariable(f"y_{c}", 'B', 0))
+
+    model.addConstraint('sum(y)', y, [1 for _ in range(len(y))], size_of_committee, 'E')
```

Comment:
1. `[1 for _ in range(len(y))]` can be replaced with `[1] * len(y)`
2. I would use a named argument or create a temporary variable for this list argument before the function call. Right now it is difficult to understand this arguement.

Reply:
2. Did you mean the list of ones? If so, I believe it to be obvious as soon as you read the 'name' argument which states 'sum(y)'